### PR TITLE
fix helm2 delete charts

### PR DESCRIPTION
**Pull request by vanajamukkara**
fix helm2 chart deletion

### CODE REVIEW

File: src/integration-tests/bash/cleanup.sh [hunk 1]
```diff
@@ -524,31 +524,34 @@ if [ -x "$(command -v helm)" ]; then
...
+        set -x
+        helm delete --purge  $helm_name
+        )
+      else
+        echo @@ `timestamp` Info: DRYRUN: helm delete --purge  $helm_name
+      fi
+    done
+  else
+    namespaces=`kubectl get ns | grep -v NAME | awk '{ print $1 }'`
+    for ns in $namespaces
```
Comment (markxnelson) [inline src/integration-tests/bash/cleanup.sh]: still don't need this loop....
`helm list --all-namespaces`

Comment (vanajamukkara) [reply src/integration-tests/bash/cleanup.sh]: yes, I haven't fixed helm3 ..may be @anpanigr or I can look into helm3. I have only fixed helm2 for now, external jenkins is using helm2. Have something to work on mii, will get back to helm3 later..

Comment (anpanigr) [reply src/integration-tests/bash/cleanup.sh]: Helm 3 need an explicit namespace to delete helm chart. So we need to loop thru the same space.  If we can find the list of name spaces dynamically created for the testing, we can avoid looping for all namesapces

Comment (vanajamukkara) [reply src/integration-tests/bash/cleanup.sh]: @anpanigr how about do `helm list --all-namespaces` and get the ns and use that for deletion instead of looping through all the existing namespaces..

Comment (tbarnes-us) [reply src/integration-tests/bash/cleanup.sh]: Why is  this such a big worry? The script takes several seconds to run, and this would only shave off milliseconds.

Comment (tbarnes-us) [reply src/integration-tests/bash/cleanup.sh]: (Just curious if I'm misunderstanding.)

Comment (markxnelson) [reply src/integration-tests/bash/cleanup.sh]: yesterday i watched it take over 15 minutes as it ran helm list 483 times

Comment (markxnelson) [reply src/integration-tests/bash/cleanup.sh]: sure it will run in a few seconds if there is nothing to delete!

Comment (vanajamukkara) [reply src/integration-tests/bash/cleanup.sh]: more namespaces more time ..

Comment (rjeberhard) [reply src/integration-tests/bash/cleanup.sh]: You should be able to get the namespace from the table.  If that is hard to parse, `helm list` supports output formats of json and yaml.

Comment (tbarnes-us) [reply src/integration-tests/bash/cleanup.sh]: Oh. That's an impressive namespaces count!

Comment (vanajamukkara) [reply src/integration-tests/bash/cleanup.sh]: yes. helm3 chart deletion needs to be reviewed and fixed. In any case, this PR is just for helm2 fix for now. I have submitted a job on external jenkins with my branch. @markxnelson @rjeberhard you can merge this PR if you are comfortable after seeing the jenkins results.

**Comment by tbarnes-us** [comment]
I realize it's unrelated to this change, but we can also make the 'genericDelete' a little safer in terms of its ordering.   Change:

genericDelete "all,cm,pvc,roles,rolebindings,serviceaccount,secrets,ingress" "crd,pv,ns,clusterroles,clusterrolebindings" "logstash|kibana|elastisearch|weblogic|elk|domain|traefik|voyager|apache-webtier|mysql"

into two calls to genericDelete.  Where:

(1) we generic delete the first time with  "all" "" "logstash|kibana|elastisearch|weblogic|elk|domain|traefik|voyager|apache-webtier|mysql"

(2) we generic delete the second time with the complete list as before...

**Comment by vanajamukkara** [comment]
Quick test ran clean on external jenkins - https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-quicktest/1179/consoleFull

File: src/integration-tests/bash/cleanup.sh (hunk 1 above)
Comment (rjeberhard) [reply src/integration-tests/bash/cleanup.sh]: Why are we spending time on fixing helm2 when we dropped helm2 support for 3.0.0?  For this reason, I'm only interested in fixing helm3 and having our external Jenkins use helm3.

Comment (vanajamukkara) [reply src/integration-tests/bash/cleanup.sh]: @anpanigr changed the integration tests for helm3, but the changes were not tested on external jenkins using helm3. That needs to be done along with the helm3 fix in cleanup.sh. I can close this PR if we don't need helm2 fix.

**Comment by vanajamukkara** [comment]
Closing the PR. New PR will be created after testing helm3 on external jenkins, with helm3 fix in cleanup script.

File: src/integration-tests/bash/cleanup.sh (hunk 1 above)
Comment (markxnelson) [reply src/integration-tests/bash/cleanup.sh]: ok - well let's make sure the test are in fact only using helm 3 then please :)
